File: statsig-dotnet/dotnet-statsig/Experiment.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Statsig
{
    public class Experiment
    {
        [JsonProperty("name")] public string Name;
        [JsonProperty("rule_id")] public string RuleId;
        [JsonProperty("value")] public IReadOnlyDictionary<string, JToken> Value;
        [JsonProperty("group_name")] public string GroupName;
        [JsonProperty("details")] public EvaluationDetails Details;

        internal Experiment(string name, string ruleId, IReadOnlyDictionary<string, JToken> value, string groupName, EvaluationDetails details)
        {
            Name = name;
            RuleId = ruleId;
            Value = value;
            GroupName = groupName;
            Details = details;
        }

        // empty constructor when deserializing objects
        internal Experiment()
        {
        }
    }
}

File: statsig-java/src/main/java/com/statsig/Experiment.java
package com.statsig;

import com.alibaba.fastjson2.JSON;
import com.alibaba.fastjson2.JSONObject;
import com.alibaba.fastjson2.TypeReference;
import com.alibaba.fastjson2.annotation.JSONCreator;
import com.alibaba.fastjson2.annotation.JSONField;
import java.util.List;
import java.util.Map;

public class Experiment extends BaseConfig {
  public final String groupName;
  public List<Map<String, String>> secondaryExposures;

  @JSONCreator
  Experiment(
      @JSONField(name = "name") String name,
      @JSONField(name = "value") Map<String, Object> value,
      @JSONField(name = "ruleID") String ruleID,
      @JSONField(name = "details") EvaluationDetails evaluationDetails,
      @JSONField(name = "idType") String idType,
      @JSONField(name = "groupName") String groupName,
      @JSONField(name = "secondaryExposures") List<Map<String, String>> secondaryExposures) {
    super(name, value, ruleID, evaluationDetails, idType);
    this.groupName = groupName;
    this.secondaryExposures = secondaryExposures;
  }

  public String getGroupName() {
    return groupName;
  }

  public List<Map<String, String>> getSecondaryExposures() {
    return secondaryExposures;
  }

  static Experiment fromJson(String json) {
    if (json == null || json.isEmpty()) {
      return null;
    }

    try {
      Experiment experiment = JSON.parseObject(json, Experiment.class);
      if (experiment == null) {
        return null;
      }

      experiment.setRawJson(json);

      JSONObject root = JSON.parseObject(json);
      if (root != null) {
        Object rawSecondaryExposures = root.get("secondaryExposures");
        if (rawSecondaryExposures != null) {
          List<Map<String, String>> se =
              JSON.parseObject(
                  rawSecondaryExposures.toString(),
                  new TypeReference<List<Map<String, String>>>() {});
          experiment.secondaryExposures = se;
        }
      }
      return experiment;
    } catch (Exception e) {
      System.err.println("Error deserializing Experiment: " + e.getMessage());
      e.printStackTrace();
      return null;
    }
  }
}
